src/components/DatasetDetailView.jsx

```diff
@@ -76,11 +77,15 @@ export class DatasetDetailView extends React.PureComponent {
   }
 
   render() {
-    const { classes, dataset, datasetPolicies, dispatch } = this.props;
+    const { classes, dataset, datasetPolicies, dispatch, userEmail } = this.props;
     const datasetReadersObj = datasetPolicies.find(policy => policy.name === 'reader'); // TODO make this an enum
     const datasetReaders = (datasetReadersObj && datasetReadersObj.members) || [];
     const datasetCustodiansObj = datasetPolicies.find(policy => policy.name === 'custodian');
     const datasetCustodians = (datasetCustodiansObj && datasetCustodiansObj.members) || [];
+    const datasetStewardsObj = datasetPolicies.find(policy => policy.name === 'steward');
+    const datasetStewards = (datasetStewardsObj && datasetStewardsObj.members) || [];
+    const allowedToAddReaders =
+      _.includes(datasetStewards, userEmail) || _.includes(datasetCustodians, userEmail);
```

Comment:
This won't work. We store the stewards group in the policy, not the individual steward emails. We should really be doing this by asking SAM if the user is allowed to perform the `share_policy` action on the dataset.

Reply:
This is fine for now. I added [this as a follow-on.](https://broadworkbench.atlassian.net/browse/DR-392)